CMakeLists.txt

```diff
@@ -19,72 +23,106 @@ IF ("${CMAKE_SOURCE_DIR}" STREQUAL "${CMAKE_BINARY_DIR}")
+  IF(("${CMAKE_VERSION}" VERSION_GREATER 3.0) OR("${CMAKE_VERSION}" VERSION_EQUAL 3.0))
+    CMAKE_POLICY(SET CMP0026 OLD) # Read LOCATION properties from build targets
+    CMAKE_POLICY(SET CMP0042 OLD) # Disable @rpath in target install name
   ENDIF()
-ENDIF (COMMAND cmake_policy)
+ENDIF(COMMAND cmake_policy)
 
 # ------------------------------------------------------------------------------
 # Release information
 # ------------------------------------------------------------------------------
-SET (XCDF_MAJOR_VERSION  3 CACHE STRING "Major version number")
-SET (XCDF_MINOR_VERSION  0 CACHE STRING "Minor version number")
-SET (XCDF_PATCH_VERSION  1 CACHE STRING "Patch number")
+SET(XCDF_MAJOR_VERSION 4 CACHE STRING "Major version number")
+SET(XCDF_MINOR_VERSION 0 CACHE STRING "Minor version number")
+SET(XCDF_PATCH_VERSION 0 CACHE STRING "Patch number")
 
 # ------------------------------------------------------------------------------
-# Set up build paths 
+# Set up build paths
 # ------------------------------------------------------------------------------
-SET (CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake)
-INCLUDE (Utility)
+LIST(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
+INCLUDE(Utility)
+INCLUDE(RPathHandling)
 
-INCLUDE_DIRECTORIES (${CMAKE_SOURCE_DIR}/include)
-INCLUDE_DIRECTORIES (${CMAKE_SOURCE_DIR}/include/utility)
-INCLUDE_DIRECTORIES (${CMAKE_SOURCE_DIR}/include/alias)
-INCLUDE_DIRECTORIES (${CMAKE_SOURCE_DIR}/src/pybindings)
+INCLUDE_DIRECTORIES(${CMAKE_CURRENT_SOURCE_DIR}/include)
+INCLUDE_DIRECTORIES(${CMAKE_CURRENT_SOURCE_DIR}/include/utility)
+INCLUDE_DIRECTORIES(${CMAKE_CURRENT_SOURCE_DIR}/include/alias)
+INCLUDE_DIRECTORIES(${CMAKE_CURRENT_SOURCE_DIR}/src/pybindings)
 
 # ------------------------------------------------------------------------------
 # Set up core library and utility programs
 # ------------------------------------------------------------------------------
-XCDF_ADD_LIBRARY (TARGET xcdf
+XCDF_ADD_LIBRARY(TARGET xcdf
   HEADERS include/xcdf/*.h
-          include/xcdf/*.h.in
-          include/xcdf/utility/*.h
-          include/xcdf/alias/*.h
+  include/xcdf/*.h.in
+  include/xcdf/utility/*.h
+  include/xcdf/alias/*.h
   SOURCES src/*.cc)
 
-
+# -------------------------------------
+# Build Python bindings using pybind11
+# -------------------------------------
 OPTION(BUILD_PYTHON "Build python bindings" ON)
+
 if(BUILD_PYTHON)
-    add_subdirectory(${PROJECT_SOURCE_DIR}/extern/pybind11 )
-
-    # to avoid a naming conflict with the cpp library but still
-    # have the python module called "xcdf" we have the cmake target pyxcdf
-    # and set its outputname to xcdf
-    pybind11_add_module(pyxcdf python/xcdf.cpp)
-    set_target_properties(pyxcdf PROPERTIES OUTPUT_NAME xcdf)
-    target_link_libraries(pyxcdf PRIVATE xcdf)
-    target_include_directories(pyxcdf PRIVATE python)
+  add_subdirectory(${PROJECT_SOURCE_DIR}/extern/pybind11)
```

Comment:
Do not use both `add_subdirectory` and `find_package` for pybind11

Reply:
by doing `find_package(pybind11 REQUIRED)` later, it already is.

Reply:
https://stackoverflow.com/a/65176396/3838691

Says to prefer `find_package`.

Reply:
Using both at the same time just doesn't make sense! If you use `add_subdirectory`,  the whole `IF(SKBUILD)` and having pybind11 in pyproject.toml doesn't make any sense.

Either use `find_package` and the whole skbuild shebang OR use `add_subdirectory`

Reply:
yes, I am suggesting to remove `find_package ` and keep `add_subdirectory `